core/src/main/scala/org/apache/spark/sql/execution/CodegenSparkFallback.scala

```diff
@@ -87,6 +88,15 @@ case class CodegenSparkFallback(var child: SparkPlan) extends UnaryExecNode {
             }
           case None => throw t
         }
+    } finally {
+      // SNAP-1422
+      val cacheMetaDataForConnector =
+        java.lang.Boolean.getBoolean("SMART_CONNECTOR_CACHE_METADATA")
+      lazy val session = sqlContext.sparkSession.asInstanceOf[SnappySession]
+      if (!cacheMetaDataForConnector &&
+        ExternalStoreUtils.isSmartConnectorMode(session.sparkContext)) {
+        session.sessionCatalog.invalidateAll()
```

Comment:
Then we should do it in getPlan too. You have a df and the corresponding table is deleted and now you do df.collect. This case will be handled in CodeGenSparkFallback.